restapi/src/test/java/io/stargate/web/docsapi/service/query/ExpressionParserIntTest.java

```diff
@@ -509,5 +632,213 @@ public void multipleFields() {
+                                                                                  .value()
+                                                                                  .get())
+                                                                          .isEqualTo("some-value");
+                                                                    });
+                                                            assertThat(sc.getFilterOperation())
+                                                                .isEqualTo(EqFilterOperation.of());
+                                                            assertThat(sc.getQueryValue())
+                                                                .isEqualTo("some-value");
+                                                          });
+                                                }))
+                                .anySatisfy(
+                                    c ->
+                                        assertThat(c)
+                                            .isInstanceOfSatisfying(
+                                                FilterExpression.class,
+                                                f2 -> {
+                                                  assertThat(f2.getOrderIndex()).isOne();
+                                                  assertThat(f2.getFilterPath().getField())
+                                                      .isEqualTo("myOtherField");
+                                                  assertThat(f2.getFilterPath().getParentPath())
+                                                      .isEmpty();
+                                                  assertThat(f2.getCondition())
+                                                      .isInstanceOfSatisfying(
+                                                          StringCondition.class,
+                                                          sc -> {
+                                                            assertThat(sc.getBuiltCondition())
+                                                                .isEmpty();
+                                                            assertThat(sc.getFilterOperation())
+                                                                .isEqualTo(NeFilterOperation.of());
+                                                            assertThat(sc.getQueryValue())
+                                                                .isEqualTo("some-small-value");
+                                                          });
+                                                }));
+                          }));
+    }
+
+    @Test
+    public void orNotInArray() {
+      ObjectNode root = mapper.createObjectNode();
+      root.set(
+          "$or",
+          mapper
+              .createObjectNode()
+              .set("myField", mapper.createObjectNode().put("$eq", "some-value")));
```

Comment:
Does the test name match actual use case here? I do not see an `IN` :thinking:

Reply:
This is a case when you have something like `"$or": {}` instead of `"$or: []`.. Will adapt..

Reply:
Done.